addons/hr_holidays/models/hr_leave.py

```diff
@@ -191,9 +191,11 @@ def _default_get_request_parameters(self, values):
         help='This area is automatically filled by the user who validate the time off with second level (If time off type need second validation)')
     can_reset = fields.Boolean('Can reset', compute='_compute_can_reset')
     can_approve = fields.Boolean('Can Approve', compute='_compute_can_approve')
+    support_documents_ids = fields.Many2many('ir.attachment', string="Attach File", inverse='_inverse_support_documents_ids')
```

Comment:
@kba-odoo , To display the attachment in chatter attachment box we have to pass the res_id and res_model. that's why the `inverse` method is used. This is new M2M field so record will not be linked automatically with bottom chatter attachment. 

We may use `one2many` with res_id of attachment but it will link all the chatter attachment with record and display all those attachments in form view and we do not want that.  Also we do not have widget for one2many to display the attachment view same as 'many2many_binary'. 

Thanks

Reply:
@mba-odoo  Why don't we use the documents.mixin over here ?

It seems we only want to link attachment to document.documents.

Reply:
@tivisse , We have nothing to do with the documents module here. 
we want M2M with attachment in form view like: https://drive.google.com/file/d/1EJVuuvNyrfJ8d8Jvnqp_JoHxc0M3-We-/view
And that File is recorded on the chatter of the Time Off request but if any user attach the any attachment from the chatter then it should not be displayed in the form view of that Time Off Request.

Thanks